traits/tests/test_trait_dict_object.py

```diff
@@ -0,0 +1,283 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+import copy
+import pickle
+import unittest
+from unittest import mock
+
+from traits.trait_dict_object import TraitDict, TraitDictObject
+from traits.trait_errors import TraitError
+from traits.trait_types import Dict, Int, Str
+
+
+def str_validator(value):
+    if type(value) is str:
+        return value
+    else:
+        raise TraitError
+
+
+def int_validator(value):
+    if type(value) is int:
+        return value
+    else:
+        raise TraitError
+
+
+class TestTraitDict(unittest.TestCase):
+
+    def setUp(self):
+        self.added = None
+        self.changed = None
+        self.removed = None
+        self.trait_dict = None
+
+    def notification_handler(self, trait_dict, added, changed, removed):
+        self.trait_list = trait_dict
+        self.added = added
+        self.changed = changed
+        self.removed = removed
+
+    def test_init(self):
+        td = TraitDict({"a": 1, "b": 2}, key_validator=str_validator,
+                       value_validator=int_validator)
+        self.assertDictEqual(td, {"a": 1, "b": 2})
```

Comment:
Not a request change, just a note: This was changed to `assertEqual` in another PR. The two are equivalent here, and I honestly don't mind which one.

Reply:
Changed everything to use `assertEqual`